### ISSUE TITLE
gym fails with : Could not find CFPropertyList-2.3.5 in any of the sources

### ISSUE THREAD

**Issue by lakamsani** [fastlane/fastlane#10492]
### Issue Description

gym fails with : Could not find CFPropertyList-2.3.5 in any of the sources

##### Complete output when running fastlane, including the stack trace and command used

Command used:
```bash
bundle exec fastlane jenkins app:dwts --verbose
```
It runs a jenkins lane for an app called dwts. There is a ruby map that maps that short app name (dwts in this case) to a bundle Id shown in the output below. Its working fine on a few other laptops in our team and on our jenkins server for over a year or so for 4 or so different apps for jenkins/debug, adhoc and app store lanes. This problem has popped up on a brand new laptop we are trying to setup this week. 

[INSERT OUTPUT HERE]
+-------------------------------------------------------------------+-----------------------------------------------------------+
|                                                    Summary for gym 2.60.1                                                     |
+-------------------------------------------------------------------+-----------------------------------------------------------+
| workspace                                                         | Onkore.xcworkspace                                        |
| scheme                                                            | DWTSUSAiPhoneDebug                                        |
| export_method                                                     | development                                               |
| output_directory                                                  | archives                                                  |
| xcargs                                                            | PROVISIONING_PROFILE=profile DEVELOPMENT_TEAM=XXXXXXXXXX  |
| clean                                                             | true                                                      |
| include_bitcode                                                   | false                                                     |
| export_options.provisioningProfiles.com.onkore.DWTSUSA.Debug      | match Development com.onkore.DWTSUSA.Debug                |
| export_options.provisioningProfiles.com.onkore.BachelorFantasyUSA | match AppStore com.onkore.BachelorFantasyUSA              |
| export_options.provisioningProfiles.com.onkore.SharkTankUSA       | match AppStore com.onkore.SharkTankUSA                    |
| export_options.provisioningProfiles.com.onkore.SurvivorUSA        | match AppStore com.onkore.SurvivorUSA                     |
| export_options.provisioningProfiles.com.onkore.WWEUSA             | match AppStore com.onkore.WWEUSA                          |
| export_options.provisioningProfiles.com.onkore.DWTSUSA            | match AppStore com.onkore.DWTSUSA                         |
| export_options.provisioningProfiles.com.onkore.ElectionsUSA       | match AppStore com.onkore.ElectionsUSA                    |
| destination                                                       | generic/platform=iOS                                      |
| output_name                                                       | Fantasy League for Dancing with the Stars                 |
| build_path                                                        | /Users/vamsee/Library/Developer/Xcode/Archives/2017-10-03 |
| silent                                                            | false                                                     |
| skip_package_ipa                                                  | false                                                     |
| buildlog_path                                                     | ~/Library/Logs/gym                                        |
| xcode_path                                                        | /Applications/Xcode.app                                   |
+-------------------------------------------------------------------+-----------------------------------------------------------+

+-----------------------------------------------------------------------------------------------+----------------------+
|                                               Generated Build Command                                                |
+-----------------------------------------------------------------------------------------------+----------------------+
| Option                                                                                        | Value                |
+-----------------------------------------------------------------------------------------------+----------------------+
| set -o pipefail &&                                                                            |                      |
| xcodebuild                                                                                    |                      |
| -workspace Onkore.xcworkspace                                                                 |                      |
| -scheme DWTSUSAiPhoneDebug                                                                    |                      |
| -destination                                                                                  | generic/platform=iOS |
| -archivePath /Users/vamsee/Library/Developer/Xcode/Archives/2017-10-03/Fantasy\ League\ for\  |                      |
| Dancing\ with\ the\ Stars\ 2017-10-03\ 22.21.02.xcarchive                                     |                      |
| PROVISIONING_PROFILE=profile DEVELOPMENT_TEAM=XXXXXXXXXX                                      |                      |
| clean                                                                                         |                      |
| archive                                                                                       |                      |
| | tee /Users/vamsee/Library/Logs/gym/Fantasy\ League\ for\ Dancing\ with\ the\                |                      |
| Stars-DWTSUSAiPhoneDebug.log                                                                  |                      |
| | xcpretty                                                                                    |                      |
+-----------------------------------------------------------------------------------------------+----------------------+
INFO [2017-10-03 22:21:02.54]: $ set -o pipefail && xcodebuild -workspace Onkore.xcworkspace -scheme DWTSUSAiPhoneDebug -destination 'generic/platform=iOS' -archivePath /Users/vamsee/Library/Developer/Xcode/Archives/2017-10-03/Fantasy\ League\ for\ Dancing\ with\ the\ Stars\ 2017-10-03\ 22.21.02.xcarchive PROVISIONING_PROFILE=profile DEVELOPMENT_TEAM=XXXXXXXXXX clean archive | tee /Users/vamsee/Library/Logs/gym/Fantasy\ League\ for\ Dancing\ with\ the\ Stars-DWTSUSAiPhoneDebug.log | xcpretty
INFO [2017-10-03 22:21:02.68]: ▸ Could not find CFPropertyList-2.3.5 in any of the sources
INFO [2017-10-03 22:21:02.68]: ▸ Run `bundle install` to install missing gems.
INFO [2017-10-03 22:24:46.96]: ▸ ** ARCHIVE FAILED **
INFO [2017-10-03 22:24:46.96]: ▸ The following build commands failed:
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/MyTournamentsViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Profile/ProfileMainViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/JoinTournamentViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/Paging/BaseTournamentPageViewcontroller.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Performers/PerformersViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/InAppMessages/IamDynamicTourneyInviteViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tiles/ContestantLeaderboardsViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controls/QuestionView/QuestionViewItems.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/BaseTournamentViewcontroller.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/DataControllers/ChatDataController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Debug/DebugMessagesViewController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/DataControllers/TournamentDataController.swift
INFO [2017-10-03 22:24:46.96]: ▸ CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
INFO [2017-10-03 22:24:46.96]: ▸ (13 failures)
Could not find CFPropertyList-2.3.5 in any of the sources
Run `bundle install` to install missing gems.
** ARCHIVE FAILED **

The following build commands failed:
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/MyTournamentsViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Profile/ProfileMainViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/JoinTournamentViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/Paging/BaseTournamentPageViewcontroller.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Performers/PerformersViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/InAppMessages/IamDynamicTourneyInviteViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tiles/ContestantLeaderboardsViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controls/QuestionView/QuestionViewItems.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Tournaments/BaseTournamentViewcontroller.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/DataControllers/ChatDataController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/Debug/DebugMessagesViewController.swift
CompileSwift normal arm64 /Users/vamsee/Development/OnKore/onkore-ios/iPhone/Onkore/Source/Controllers/DataControllers/TournamentDataController.swift
CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(13 failures)
ERROR [2017-10-03 22:24:47.00]: Exit status: 7

+---------------+-------------------------+
|            Build environment            |
+---------------+-------------------------+
| xcode_path    | /Applications/Xcode.app |
| gym_version   | 2.60.1                  |
| export_method | development             |
| sdk           | iPhoneOS11.0.sdk        |
+---------------+-------------------------+

WARN [2017-10-03 22:24:47.72]: Lane Context:
INFO [2017-10-03 22:24:47.72]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios jenkins", :SIGH_PROFILE_TYPE=>"development", :MATCH_PROVISIONING_PROFILE_MAPPING=>{"com.onkore.DWTSUSA.Debug"=>"match Development com.onkore.DWTSUSA.Debug"}}
ERROR [2017-10-03 22:24:47.72]: undefined method `each' for nil:NilClass
INFO [2017-10-03 22:24:47.72]: Successfully generated documentation at path '/Users/vamsee/Development/OnKore/onkore-ios/fastlane/README.md'

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | cocoapods                           | 8           |
| 4    | match                               | 11          |
| 💥   | gym                                 | 229         |
+------+-------------------------------------+-------------+

DEBUG [2017-10-03 22:24:47.74]: All plugins are up to date
ERROR [2017-10-03 22:24:47.74]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `each' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60each'%20for%20nil+repo:fastlane/fastlane
➡️  fastlane supply fails with error "NoMethodError: [!] undefined method `map' for nil:NilClass"
    https://github.com/fastlane/fastlane/issues/10033 [closed] 14 💬
    3 weeks ago

➡️  undefined method `id' for nil:NilClass while fastlane sigh repair
    https://github.com/fastlane/fastlane/issues/8184 [closed] 5 💬
    a week ago

➡️  undefined method `fetch' for nil:NilClass in spaceship through deliver
    https://github.com/fastlane/fastlane/issues/9373 [closed] 4 💬
    4 weeks ago

and 102 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60each%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
DEBUG [2017-10-03 22:24:48.41]: All plugins are up to date
bundler: failed to load command: fastlane (/Users/vamsee/Development/ruby/gems/ruby/2.3.0/bin/fastlane)
NoMethodError: [!] undefined method `each' for nil:NilClass
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/error_handler.rb:144:in `print_full_log_path'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/error_handler.rb:56:in `handle_build_error'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/runner.rb:90:in `block in build_app'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane_core/lib/fastlane_core/command_executor.rb:88:in `execute'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/runner.rb:86:in `build_app'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/runner.rb:11:in `run'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/gym/lib/gym/manager.rb:14:in `work'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/actions/gym.rb:65:in `run'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:252:in `block (2 levels) in execute_action'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:230:in `block in execute_action'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:226:in `chdir'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:226:in `execute_action'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
  Fastfile:53:in `block (2 levels) in parsing_binding'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:64:in `run!'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/fastlane/lib/fastlane/cli_tools_distributor.rb:66:in `take_off'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/gems/fastlane-2.60.1/bin/fastlane:20:in `<top (required)>'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/bin/fastlane:22:in `load'
  /Users/vamsee/Development/ruby/gems/ruby/2.3.0/bin/fastlane:22:in `<top (required)>'

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

### Captured Output

Command Used: `env`
<details><summary>Output/Log</summary>

```

-------
WARN [2017-10-03 22:34:09.65]: Generating fastlane environment output, this might take a few seconds...
DEBUG [2017-10-03 22:34:09.88]: Checking if there are any plugins that should be loaded...
DEBUG [2017-10-03 22:34:09.89]: Loading 'fastlane-plugin-upload_symbols_to_new_relic' plugin
INFO [2017-10-03 22:34:10.00]: gem 'fastlane-plugin-upload_symbols_to_new_relic' is already installed
DEBUG [2017-10-03 22:34:10.00]: Loading 'fastlane-plugin-appicon' plugin
INFO [2017-10-03 22:34:10.11]: gem 'fastlane-plugin-appicon' is already installed
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13                                                          |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.13.5 (Apple Git-94)                              |
| Installation Source         | ~/Development/ruby/gems/ruby/2.3.0/bin/fastlane                |
| Host                        | Mac OS X 10.13 (17A365)                                        |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.0                                                            |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.60.1"

$:.unshift File.dirname(__FILE__)
require 'Onkore.rb'

default_platform :ios

platform :ios do
  before_all do
    ENV["FASTLANE_XCODE_LIST_TIMEOUT"] = "120"
    Onkore.setupEnv  
    if ENV['FL_CHECKOUT']
        puts "Checking out code to ensure build directory is clean ..."
        sh("git checkout ../")
    end
    ensure_git_status_clean unless ENV['FL_NO_ENSURE_CLEAN']
    ENV["FASTFILE_DIR"] =  File.expand_path(File.dirname(File.dirname(__FILE__)))
  end


  desc "Parameterized Jenkins lane. If params are omitted will default to Bachelor"
  lane :jenkins do |options|
    cocoapods unless ENV['FL_NO_PODS']
    type = 'development'
    app = Onkore.getApp(options,type)
    match(app_identifier:app['appid'] , readonly: true, type: type)
    suffixRec = Onkore.getSuffix(type)
    profile= suffixRec['profile'] + app['appid']
    team=ENV['FASTLANE_TEAM_ID']

    gym(
        workspace: 'Onkore.xcworkspace',
        scheme: app['scheme'],
        export_method: "development",
        output_directory: "archives",
        xcargs: "PROVISIONING_PROFILE=profile DEVELOPMENT_TEAM=#{team}",
        clean: true, include_bitcode:false
    )
  end

# REMOVED OTHER LANES

  after_all do |lane|
    # This block is called, only if the executed lane was successful
  end

  error do |lane, exception|
     #slack(
     #  message: exception.message,
     #  success: false
     #)
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.60.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                      | Version | Update-Status |
| ------------------------------------------- | ------- | ------------- |
| fastlane-plugin-upload_symbols_to_new_relic | 0.1.1   | ✅ Up-To-Date  |
| fastlane-plugin-appicon                     | 0.10.0  | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                         | Version      |
| ------------------------------------------- | ------------ |
| did_you_mean                                | 1.0.0        |
| bundler                                     | 1.15.4       |
| io-console                                  | 0.4.5        |
| CFPropertyList                              | 2.3.5        |
| i18n                                        | 0.8.6        |
| minitest                                    | 5.10.3       |
| thread_safe                                 | 0.3.6        |
| tzinfo                                      | 1.2.3        |
| activesupport                               | 4.2.10       |
| public_suffix                               | 2.0.5        |
| addressable                                 | 2.5.2        |
| babosa                                      | 1.0.2        |
| claide                                      | 1.0.2        |
| fuzzy_match                                 | 2.0.4        |
| nap                                         | 1.1.0        |
| cocoapods-core                              | 1.3.1        |
| cocoapods-deintegrate                       | 1.0.1        |
| cocoapods-downloader                        | 1.1.3        |
| cocoapods-plugins                           | 1.0.0        |
| cocoapods-search                            | 1.0.0        |
| cocoapods-stats                             | 1.0.0        |
| netrc                                       | 0.11.0       |
| cocoapods-trunk                             | 1.3.0        |
| cocoapods-try                               | 1.1.0        |
| colored2                                    | 3.1.2        |
| escape                                      | 0.0.4        |
| fourflusher                                 | 2.0.1        |
| gh_inspector                                | 1.0.3        |
| molinillo                                   | 0.5.7        |
| ruby-macho                                  | 1.1.0        |
| nanaimo                                     | 0.2.3        |
| xcodeproj                                   | 1.5.2        |
| cocoapods                                   | 1.3.1        |
| colored                                     | 1.2          |
| highline                                    | 1.7.8        |
| commander-fastlane                          | 4.4.5        |
| declarative                                 | 0.0.10       |
| declarative-option                          | 0.1.0        |
| unf_ext                                     | 0.0.7.4      |
| unf                                         | 0.1.4        |
| domain_name                                 | 0.5.20170404 |
| dotenv                                      | 2.2.1        |
| excon                                       | 0.59.0       |
| multipart-post                              | 2.0.0        |
| faraday                                     | 0.13.1       |
| http-cookie                                 | 1.0.3        |
| faraday-cookie_jar                          | 0.0.6        |
| faraday_middleware                          | 0.12.2       |
| fastimage                                   | 2.1.0        |
| jwt                                         | 1.5.6        |
| little-plugger                              | 1.1.4        |
| multi_json                                  | 1.12.2       |
| logging                                     | 2.2.2        |
| memoist                                     | 0.16.0       |
| os                                          | 0.9.6        |
| signet                                      | 0.7.3        |
| googleauth                                  | 0.5.3        |
| httpclient                                  | 2.8.3        |
| mime-types-data                             | 3.2016.0521  |
| mime-types                                  | 3.1          |
| uber                                        | 0.1.0        |
| representable                               | 3.0.4        |
| retriable                                   | 3.1.1        |
| google-api-client                           | 0.13.6       |
| json                                        | 2.1.0        |
| mini_magick                                 | 4.5.1        |
| multi_xml                                   | 0.6.0        |
| plist                                       | 3.3.0        |
| rubyzip                                     | 1.2.1        |
| security                                    | 0.1.3        |
| slack-notifier                              | 1.5.1        |
| terminal-notifier                           | 1.8.0        |
| unicode-display_width                       | 1.3.0        |
| terminal-table                              | 1.8.0        |
| tty-screen                                  | 0.5.0        |
| word_wrap                                   | 1.0.0        |
| rouge                                       | 2.0.7        |
| xcpretty                                    | 0.2.8        |
| xcpretty-travis-formatter                   | 0.0.4        |
| fastlane-plugin-appicon                     | 0.10.0       |
| fastlane-plugin-upload_symbols_to_new_relic | 0.1.1        |
| xcode-install                               | 2.3.0        |
</details>

*generated on:* **2017-10-03**
</details>
INFO [2017-10-03 22:34:10.59]: Open https://github.com/fastlane/fastlane/issues/new to submit a new issue ✅

```

</details>

**Comment by lakamsani** [comment]
Output of `bundle install`

````bash
onkore-ios git:(develop) ✗ bundle install
The latest bundler is 1.16.0.pre.2, but you are currently running 1.15.4.
To update, run `gem install bundler --pre`
Fetching gem metadata from https://rubygems.org/..............
Fetching version metadata from https://rubygems.org/...
Fetching dependency metadata from https://rubygems.org/..
Resolving dependencies...
Using CFPropertyList 2.3.5
Using i18n 0.8.6
Using minitest 5.10.3
Using thread_safe 0.3.6
Using public_suffix 2.0.5
Using babosa 1.0.2
Using bundler 1.15.4
Using claide 1.0.2
Using fuzzy_match 2.0.4
Using nap 1.1.0
Using cocoapods-deintegrate 1.0.1
Using cocoapods-downloader 1.1.3
Using cocoapods-search 1.0.0
Using cocoapods-stats 1.0.0
Using netrc 0.11.0
Using cocoapods-try 1.1.0
Using colored2 3.1.2
Using escape 0.0.4
Using fourflusher 2.0.1
Using gh_inspector 1.0.3
Using molinillo 0.5.7
Using ruby-macho 1.1.0
Using nanaimo 0.2.3
Using colored 1.2
Using highline 1.7.8
Using declarative 0.0.10
Using declarative-option 0.1.0
Using unf_ext 0.0.7.4
Using dotenv 2.2.1
Using excon 0.59.0
Using multipart-post 2.0.0
Using fastimage 2.1.0
Using jwt 1.5.6
Using little-plugger 1.1.4
Using multi_json 1.12.2
Using memoist 0.16.0
Using os 0.9.6
Using httpclient 2.8.3
Using mime-types-data 3.2016.0521
Using uber 0.1.0
Using retriable 3.1.1
Using json 2.1.0
Using mini_magick 4.5.1
Using multi_xml 0.6.0
Using plist 3.3.0
Using rubyzip 1.2.1
Using security 0.1.3
Using slack-notifier 1.5.1
Using terminal-notifier 1.8.0
Using unicode-display_width 1.3.0
Using tty-screen 0.5.0
Using word_wrap 1.0.0
Using rouge 2.0.7
Using fastlane-plugin-upload_symbols_to_new_relic 0.1.1
Using tzinfo 1.2.3
Using addressable 2.5.2
Using cocoapods-plugins 1.0.0
Using cocoapods-trunk 1.3.0
Using xcodeproj 1.5.2
Using commander-fastlane 4.4.5
Using unf 0.1.4
Using faraday 0.13.1
Using logging 2.2.2
Using mime-types 3.1
Using representable 3.0.4
Using fastlane-plugin-appicon 0.10.0
Using terminal-table 1.8.0
Using xcpretty 0.2.8
Using activesupport 4.2.10
Using domain_name 0.5.20170404
Using faraday_middleware 0.12.2
Using signet 0.7.3
Using xcpretty-travis-formatter 0.0.4
Using cocoapods-core 1.3.1
Using http-cookie 1.0.3
Using googleauth 0.5.3
Using cocoapods 1.3.1
Using faraday-cookie_jar 0.0.6
Using google-api-client 0.13.6
Using fastlane 2.60.1
Using xcode-install 2.3.0
Bundle complete! 5 Gemfile dependencies, 81 gems now installed.
Bundled gems are installed into /Users/vamsee/Development/ruby/gems.
````

**Comment by KrauseFx** [comment]
Can you share your Gemfile? Also try `gem cleanup`

**Comment by lakamsani** [comment]
`gem cleanup` doesn't seem to make much difference. 

** Gemfie **
```ruby
source "https://rubygems.org"

gem 'fastlane'
gem "xcode-install"
gem 'cocoapods'

plugins_path = File.join(File.dirname(__FILE__), 'fastlane', 'Pluginfile')
eval(File.read(plugins_path), binding) if File.exist?(plugins_path)
```

**Pluginfile**

```ruby
gem 'fastlane-plugin-upload_symbols_to_new_relic'
gem 'fastlane-plugin-appicon'
```

**Comment by KrauseFx** [comment]
Could you try removing the plugins and see if that works for you?

**Comment by lakamsani** [comment]
Doesn't look like plugins are it. Problem may be due to ruby version. On the new mac ruby version is 2.4.0

````bash
%  ruby --version
ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin17]
```
But if I add `ruby '2.4.0'` to `Gemfile` I get this error: 

Your Ruby version is 2.3.3, but your Gemfile specified 2.4.0

As far as I can tell there is only the 2.4.0 version of ruby on this new laptop.  Do  I need to install 2.3.0 version of ruby and retry? All the Gems seems to be installed under a 2.3.0 directory under GEM_HOME.

**Comment by KrauseFx** [comment]
Do you have an easy way to try Ruby 2.3.0? Because your Gemfile looks like it's setup correctly

**Comment by lakamsani** [comment]
used `rbenv` to switch to 2.3.0 and now it is complaining about some other missing gem.

Could not find unf_ext-0.0.7.4 in any of the sources

However, it is there:

```bash
onkore-ios git:(develop) ✗ bundle info unf_ext
The latest bundler is 1.16.0.pre.2, but you are currently running 1.15.4.
To update, run `gem install bundler --pre`
  * unf_ext (0.0.7.4)
	Summary: Unicode Normalization Form support library for CRuby
	Homepage: https://github.com/knu/ruby-unf_ext
	Path: /Users/vamsee/Development/ruby/gems/unf_ext-0.0.7.4
```

**Comment by lakamsani** [comment]
It is working when I install FL directly via brew and skip bundler/Gemfiles. Actually had to delete the Gemfile and Gemfile.lock as their existence was casing the direct invoked FL to stop with new errors. 

We are not a Ruby shop (node/JS, Swift, Java and some python) but picked up enough of Ruby to work with Fastlane. We have this one laptop not using bundler but the others and our Jenkins server using bundler.  As we get more dev laptops setup we need to figure out which approach is easy to set up. 

Thanks for you feedback. I suppose we close this and try and debug this bundler issue separately.

**Comment by jdee** [comment]
I have a couple of little pieces of advice, but I agree with @KrauseFx that this looks like a Ruby environment issue. Most experienced Ruby devs will tell you to use rbenv, chruby or RVM to avoid these problems. And the Homebrew build of Fastlane is designed to solve them as well.

Regarding your original issue though. I'm not an rbenv expert (I use RVM), but you can verify this:

- Be sure to source the script generated by rbenv in order to let it manage the Ruby environment.
- Be sure that rbenv is managing all your gems, particularly the bundler. (And update RubyGems while your'e at it.)

```bash
gem update --system
gem install bundler # will update if already present
bundle install
```

(I don't think you use `sudo` when installing gems with rbenv, but I'm not entirely sure.) If somehow you don't automatically get the bundler when you install a new Ruby with rbenv, you may still be using the system bundler, and you could easily get this kind of problem. You can also check these things using `which ruby`, `which gem` and `which bundle`.

If you want to give RVM a try, I can offer more advice: https://github.com/BranchMetrics/fastlane-plugin-branch/wiki/RVM-Cheat-Sheet. Not everyone loves it, but it does work. 

I've had issues using plugins with the Homebrew build of fastlane. But if that's working for you, Homebrew is probably your best bet.

**Comment by lakamsani** [comment]
@jdee thanks for your feedback. After I switched to the 2.3.0 version of Ruby I had to do `gem install bundle` so it installs bundle from the same location as the `rbenv` installed Ruby. I removed the brew installed fastlane cask. And my app seems to build now with via `bundle`.

I did disable plugins when I tried FL via brew so its good to get bundle working again for FL. `rbenv` does not need sudo and I picked it over `rvm` as I heard it is a bit lightweight.
